sdk/keyvault/keyvault-keys/src/localCryptography/algorithms.browser.ts

```diff
@@ -0,0 +1,174 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { isNode } from "@azure/core-http";
+import { JsonWebKey, KeyOperation } from "../keysModels";
+import { LocalCryptographyUnsupportedError } from "./models";
+
+/**
+ * This file contains the implementation of local supported algorithms, which
+ * consist of wrappers around methods already available as part of NodeJS.
+ *
+ * The goal of this file is to abstract away how we implement these algorithms,
+ * since we might be using environment-specific code or dependencies.
+ *
+ * Some algorithms can't be implemented neither in the Browser or in Node
+ * without third party dependencies or considerable development.
+ *
+ * We currently don't support any operation that requires the private data of a
+ * key. Once we implement support for the private part of JWK keys and PEM keys,
+ * we will be able to increase the support of our existing algorithms.
+ */
+
+/**
+ * @internal
+ * @ignore
+ * Abstract representation of a assertion.
+ * Assertions verify that the requirements to execute a local cryptography operation are met.
+ * @param key The JSON Web Key that will be used during the local operation.
+ * @param operationName The name of the operation, as in "encrypt", "decrypt", "sign", etc.
+ */
+export type LocalAssertion = (
+  key?: JsonWebKey,
+  operationName?: LocalCryptographyOperationName
+) => void;
+
+/**
+ * @internal
+ * @ignore
+ * The list of known assertions so far.
+ * Assertions verify that the requirements to execute a local cryptography operation are met.
+ */
+export const assertions: Record<"keyOps" | "rsa" | "nodeOnly", LocalAssertion> = {
+  /**
+   * Validates that the target local cryptography operation is allowed by the key's "keyOps" property.
+   */
+  keyOps(key?: JsonWebKey, operationName?: LocalCryptographyOperationName): void {
+    if (key && key.keyOps && !key.keyOps.includes(operationName as KeyOperation)) {
+      throw new Error(`Key does not support the ${operationName} operation`);
+    }
+  },
+  /**
+   * Validates that the key type is "RSA". For operations on RSA keys.
+   */
+  rsa(key?: JsonWebKey): void {
+    if (key && key.kty! !== "RSA" && key.kty! !== "RSA-HSM") {
+      throw new Error("Key type does not match the algorithm RSA");
+    }
+  },
+  /**
+   * Passes only when the operation is being executed in NodeJS.
+   * For operations that can only run in NodeJS.
+   */
+  nodeOnly(): void {
+    if (!isNode) {
```

Comment:
Given that this is in the file algorithms.browser.ts, do we still need the isNode check?

Reply:
Nope! I'll clean these up.